mypy/build.py

```diff
@@ -2363,6 +2357,34 @@ def find_module_and_diagnose(manager: BuildManager,
             raise CompileError(["mypy: can't find module '%s'" % id])
 
 
+def exist_added_packages(suppressed: List[str],
+                        manager: BuildManager, options: Options) -> bool:
+    """Find if there are any newly added packages that were previously suppressed.
+
+    Exclude everything not in build for follow-imports=skip.
+    """
+    for dep in suppressed:
+        if dep in manager.source_set.source_modules:
+            # We don't need to add any special logic for this. If a module
+            # is added to build, importers will be invalidated by normal mechanism.
+            continue
+        path = find_module_simple(dep, manager)
+        if not path:
+            continue
+        if (options.follow_imports == 'skip' and
+                (not path.endswith('.pyi') or options.follow_imports_for_stubs)):
+            continue
+        if path.split(os.path.sep)[-1] in ('__init__.py', '__init__.pyi'):
```

Comment:
Using `str.split` for paths can be problematic, since Windows allows both `/` and `\\` as path separators, and it can be hard to ensure that the same separator is always used. On reflection, the original test seems better, if you add a comment saying that we don't rely on the test for correctness. In my previous review I didn't realize that the test can be more general than strictly necessary. 

The original test was too lenient, which is better than being too strict like this test, and implementing a perfect test doesn't seem worth it.